### ISSUE TITLE
UMich Contact Dataset class contains invalid dataset entries, where history from one dataset is used as part of other.

### ISSUE THREAD

**Issue by DanielChaseButterfield** [lunarlab-gatech/MorphoSymm-Replication#2]
As a demonstration of this bug, I'll give an example. The total number of dataset entries in the test set (before any history length shrinks it) is 377,407. This comes from 5 different datasets. 

Therefore, if we set the history length of the dataset to 150, then each of these five dataset lose 149 data points (as they don't have a long enough history length). Therefore, the new length of the test set should be 376,662, and that's how the code in https://github.com/lunarlab-gatech/Morphology-Informed-HGNN performs.

However, this repository currently returns a test set length of 377,258, which is only 149 less than the original. What's going on here?

Inspecting the code of `umich_contact_dataset.py` reveals that all five datasets are concatenated into one file, and then the history length parameter is applied. This means that there are 596 invalid entries in the test dataset, 149 for each of the datasets after the first one. These entries consider parts of the ending sequence of the previous dataset as part of their own sequence.

This bug applies for the train and validation sets as well, meaning that we currently don't have a fair comparison between their method and ours.

**Comment by DanielChaseButterfield** [comment]
Solved with commit dc75c540a1e24fe2e6d449092aed51c3147b79b4 . Primarily involved setting up a specific dataset object for each sequence, and using ConcatDataset to put them together safely without having the issues like above.

Tests proving this works can be found in `test_create_test_dataset` and `test_create_train_val_dataset` in the `tests/test_umich_contact_dataset.py` file.
